### PR TITLE
[DOC-10949] Update Docker instructions for Cloud-first

### PR DESCRIPTION
**Pull request by mdlinville**
[DOC-10949] Update Docker instructions for Cloud-first

- Also externalize Docker instructions into a version-specific include
- Also convert HTML to Markdown

This is related to #18826 but I separated it out on purpose so that I can rebase this one on that one's work.

Sending for initial review before backporting to more versions

### Previews
[src/current/v24.2/install-cockroachdb-mac.md](https://deploy-preview-18839--cockroachdb-docs.netlify.app/docs/v24.2/install-cockroachdb-mac.html)
[src/current/v24.2/install-cockroachdb-windows.md](https://deploy-preview-18839--cockroachdb-docs.netlify.app/docs/v24.2/install-cockroachdb-windows.html)
[src/current/v24.2/install-cockroachdb-linux.md](https://deploy-preview-18839--cockroachdb-docs.netlify.app/docs/v24.2/install-cockroachdb-linux.html)

### CODE REVIEW

File: src/current/_includes/v24.2/install-docker-steps.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+{{site.data.alerts.callout_danger}}
+Running a stateful application like CockroachDB in Docker is more complex and error-prone than most uses of Docker. Unless you are very experienced with Docker, we recommend starting with a different installation and deployment method.
+{{site.data.alerts.end}}
+
+CockroachDB's Docker images are [multi-platform images](https://docs.docker.com/build/building/multi-platform/) that contain binaries for both Intel and ARM. Multi-platform images do not take up additional space on your Docker host.
+
+Experimental images are not qualified for production use and not eligible for support or uptime SLA commitments.
+
+1. Install a container runtime, such as [Docker Desktop](https://docs.docker.com/desktop/install/).
```
Comment (rmloveland) [inline src/current/_includes/v24.2/install-docker-steps.md:17]: the link to Docker Desktop is 404ing

Screenshot 2024-08-20 at 11 15 30 AM

File: src/current/_includes/v24.2/install-docker-steps.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+1. Verify that the runtime service is installed correctly and running in the background. Refer to the runtime's documentation. For Docker, start a terminal and run `docker version`. If you get an error, verify your installation and try again.
+1. Visit [Docker Hub](https://hub.docker.com/r/cockroachdb/cockroach) and decide which image tag to pull. CockroachDB images are published on [Docker Hub](https://hub.docker.com/r/cockroachdb/cockroach). Releases are rolled out gradually. Docker images for a new release are published when other binary artifacts are published. The following tag formats are commonly used, although other tags are available.
```
Comment (rmloveland) [inline src/current/_includes/v24.2/install-docker-steps.md:19]: not strictly necessary to repeat the Docker Hub link IMO

File: src/current/_includes/v24.2/install-docker-steps.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    <table markdown="1">
+    <thead>
+      <tr>
+        <td>Tag</td>
+        <td>Example></td>
```
Comment (rmloveland) [inline src/current/_includes/v24.2/install-docker-steps.md:25]: This renders as `Example>` with an unnecessary `>` after the word

File: src/current/_includes/v24.2/install-docker-steps.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      </tr>
+      <tr>
+        <td>Latest patch within a major version</td>
+        <td>`latest-{{ page.version.version }}`</td>
+        <td>Automatically updates a cluster to the latest patch of the version you specify. This tag is recommended in production, because it keeps your cluster updated within a major version but does not automatically upgrade your cluster to a new major version.</td>
+      </tr>
+      <tr>
+        <td>`latest`</td>
+        <td>The latest patch within the latest major version.
+        <td>This is the default default if you do not specify a tag. It updates your cluster automatically to each new patch and major version, and is not recommended in production.</td>
```
Comment (rmloveland) [inline src/current/_includes/v24.2/install-docker-steps.md:43]: doubled word: "default default"

**Review by rmloveland — APPROVED** [review]
This LGTM

Suggest getting a cursory review from someone in #release-engineering so they're informed

File: src/current/_includes/v24.2/install-docker-steps.md (hunk 1 above)
Comment (mdlinville) [inline src/current/_includes/v24.2/install-docker-steps.md:17]:
```suggestion
1. Install a container runtime, such as [Docker Desktop](https://www.docker.com/products/docker-desktop/).
```

File: src/current/_includes/v24.2/install-docker-steps.md (hunk 2 above)
Comment (mdlinville) [inline src/current/_includes/v24.2/install-docker-steps.md:19]:
```suggestion
1. Visit [Docker Hub](https://hub.docker.com/r/cockroachdb/cockroach) and decide which image tag to pull.  Releases are rolled out gradually. Docker images for a new release are published when other binary artifacts are published. The following tag formats are commonly used; other tags are available.
```

File: src/current/_includes/v24.2/install-docker-steps.md (hunk 3 above)
Comment (mdlinville) [inline src/current/_includes/v24.2/install-docker-steps.md:25]:
```suggestion
        <td>Example</td>
```
Took some squinting!

File: src/current/_includes/v24.2/install-docker-steps.md (hunk 4 above)
Comment (mdlinville) [inline src/current/_includes/v24.2/install-docker-steps.md:43]:
```suggestion
        <td>This is the default if you do not specify a tag. It updates your cluster automatically to each new patch and major version, and is not recommended in production.</td>
```

**Pull request merged**